pkg/server/plugin/datastore/sql/sql_test.go

```diff
@@ -815,52 +816,121 @@ func (s *PluginSuite) TestFetchAttestedNodesWithPagination() {
 }
 
 func (s *PluginSuite) TestUpdateAttestedNode() {
+	userial := "deadbeef"
+	uexpires := time.Now().Add(time.Hour * 2).Unix()
+	unewSerial := ""
+	unewExpires := int64(0)
+
 	node := &common.AttestedNode{
 		SpiffeId:            "foo",
 		AttestationDataType: "aws-tag",
 		CertSerialNumber:    "badcafe",
 		CertNotAfter:        time.Now().Add(time.Hour).Unix(),
+		NewCertSerialNumber: "newbadcafe",
+		NewCertNotAfter:     time.Now().Add(2 * time.Hour).Unix(),
 	}
 
-	userial := "deadbeef"
-	uexpires := time.Now().Add(time.Hour * 2).Unix()
-
-	// update non-existing attested node
-	_, err := s.ds.UpdateAttestedNode(ctx, &datastore.UpdateAttestedNodeRequest{
-		SpiffeId:         node.SpiffeId,
-		CertSerialNumber: userial,
-		CertNotAfter:     uexpires,
-	})
-	s.RequireGRPCStatus(err, codes.NotFound, _notFoundErrMsg)
-
-	_, err = s.ds.CreateAttestedNode(ctx, &datastore.CreateAttestedNodeRequest{Node: node})
-	s.Require().NoError(err)
-
-	uresp, err := s.ds.UpdateAttestedNode(ctx, &datastore.UpdateAttestedNodeRequest{
-		SpiffeId:         node.SpiffeId,
-		CertSerialNumber: userial,
-		CertNotAfter:     uexpires,
-	})
-	s.Require().NoError(err)
-
-	unode := uresp.Node
-	s.Require().NotNil(unode)
+	fullUpdatedNode := &common.AttestedNode{
+		SpiffeId:            node.SpiffeId,
+		AttestationDataType: "aws-tag",
+		CertSerialNumber:    userial,
+		CertNotAfter:        uexpires,
+		NewCertNotAfter:     unewExpires,
+		NewCertSerialNumber: unewSerial,
+	}
 
-	s.Equal(node.SpiffeId, unode.SpiffeId)
-	s.Equal(node.AttestationDataType, unode.AttestationDataType)
-	s.Equal(userial, unode.CertSerialNumber)
-	s.Equal(uexpires, unode.CertNotAfter)
+	for _, tt := range []struct {
+		name           string
+		updateReq      *datastore.UpdateAttestedNodeRequest
+		existentNode   *common.AttestedNode
+		expUpdatedNode *common.AttestedNode
+		expErr         error
+	}{
+		{
+			name: "update non-existing attested node",
+			updateReq: &datastore.UpdateAttestedNodeRequest{
+				SpiffeId:         node.SpiffeId,
+				CertSerialNumber: userial,
+				CertNotAfter:     uexpires,
+			},
+			expErr: status.Error(codes.NotFound, _notFoundErrMsg),
+		},
+		{
+			name: "update attested node with all false mask",
+			updateReq: &datastore.UpdateAttestedNodeRequest{
+				SpiffeId:            node.SpiffeId,
+				CertSerialNumber:    userial,
+				CertNotAfter:        uexpires,
+				NewCertNotAfter:     unewExpires,
+				NewCertSerialNumber: unewSerial,
+				InputMask:           &common.AttestedNodeMask{},
+			},
+			existentNode:   node,
```

Comment:
is not it will always true? because we are veriting the same iinstance,
I think you must create a new instance for expUpdatedNode in all tests, just to be sure what we are expecting in all cases, easier to read and easier to know our result